Modernize syntax, loop handling and add handler

stejoo opened the pull request:
First of all: thanks for your code! :+1: 

I performed some maintenance on it by:
- adapted to the more modern syntax
- replaced the to-be-deprecated `with_` loop statements with the newer `loop` statements
- added a handler instead of a strictly boolean defined restart of `systemd-networkd` to improve idempotency.

Functionally this does not change your code (except for taking out unnecessary restarts), just brings the code more up to date. Which is why I made this pull request to contribute back and help this role into the future.

I plan on making other changes (in my fork) in the future that add functionality and might reorganize the structure a bit. If you are interested in those changes please let me know.

stejoo committed:
Modernize syntax and loop logic

stejoo committed:
Use a handler to restart systemd-networkd when configuration changes.

This adds a handler that is called when a config is changed, so
systemd-networkd can be restarted to apply the new configuration.
Without losing the conditional to enable/disable restarting of the
systemd-networkd service.

stejoo committed:
Merge branch 'use_handler'

aruhier commented:
Hi @stejoo,
Thanks a lot for this PR, I really appreciate! Everything looked good to me, I merged it.
Sure, if you plan any new feature, please let me know and I'll see if I can/want to merge it here!

stejoo commented:
Hi, thanks :+1: 
I'm currently working on changing things in the `develop` branch of my fork. Feel free to have a look at it. For now I changed:
- the task structure a bit
- overhauled the README
- added a bit of control over the services at boottime

I plan to:

- add a template for `resolved.conf` 
- a optional task list to disables other networking services that might conflict (like the out of the box `networking` service of Debian
